src/main/java/servlets/LoginServlet.java

```diff
@@ -26,19 +27,46 @@
 
 @WebServlet("/Login")
 public class LoginServlet extends HttpServlet {
-
+  private static final Logger LOGGER = Logger.getLogger(LoginServlet.class.getName());
   private static final String CONTENT_TYPE_TEXT_HTML = "text/html;";
   private static final String REDIRECT_LINK = "/Login";
+  private final String CLIENT_ID = "client_id=150737768611-svndjtlklolq53g4ass4r3sqal2i31p5.apps.googleusercontent.com";
+  private final String SCOPE = "scope=https://www.googleapis.com/auth/spreadsheets";
+  private final String REDIRECT_URI = "redirect_uri=";
+  private final String OAUTH_LOGIN_URI = "https://accounts.google.com/o/oauth2/v2/auth";
+  private final String RESPONSE_TYPE = "response_type=code";
+  private final String OAUTH_CALLBACK_SERVLET = "/api/oauth/callback/sheets";
+  private final String ENVIRONMENT_VARIABLE = "DOMAIN";
 
-  // Returns  a URL to either login or logout.
+  // Returns a URL to either login and get OAuth tokens or to logout.
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType(CONTENT_TYPE_TEXT_HTML);
     PrintWriter out = response.getWriter();
     if (UserAuthUtil.isUserLoggedIn()) {
       out.println(UserAuthUtil.getLogoutURL(REDIRECT_LINK));
     } else {
-      out.println(UserAuthUtil.getLoginURL(REDIRECT_LINK));
+      // Send user to OAuth consent page.
+      response.sendRedirect(getOAuthRedirectURL());
+    }
+  }
+
+  /** Build the OAuth consent page redirect URL. */
+  private String getOAuthRedirectURL() {
+    String redirectUri = REDIRECT_URI + getRedirectUri();
+    
+    return String.format("%s?%s&%s&%s&%s", OAUTH_LOGIN_URI, CLIENT_ID,
```

Comment:
Just to double check -- is this the right format?

Looks like it will produce
login.uri?clientID&responseType&redirectURI&scope

Shoud it produce something like login.uri?clientId=clientID&responseType=responseType....?

(This is a question, not a declaration that the latter is correct)

Reply:
The current implementation produces the second example. Look closely at the variables. They follow this format variableName = "variableName=varaibleContents".

Reply:
Gotcha -- sorry for the bad question!